### ISSUE TITLE
attention_mask 为全 1 时 causal_mask 为 None，导致训练出现问题

### ISSUE THREAD

**Issue by zyc140345** [GCYZSL/MoLA#21]
## 问题原因
在加载模型时，from_pretrained 方法会调用 transformers/modeling_utils.py 中的 _autoset_attn_implementation 方法自动开启 sdpa (新版 pytorch 中 Scaled Dot-Product Attention 的高效实现)：
![_autoset_attn_implementation](https://github.com/user-attachments/assets/2b5dd47f-8d44-4747-9091-83c0e66aacc0)

在 mola_modeling_llama_hacked.py 中，LlamaModel 的 forward 方法会调用 _update_causal_mask 方法更新 causal_mask，该方法会进一步调用 transformers/modeling_attn_mask_utils.py 中的 AttentionMaskConverter._ignore_causal_mask_sdpa 静态方法检测是否需要忽略 causal_mask：
![_update_causal_mask](https://github.com/user-attachments/assets/4135c6a9-64e6-4e6e-99d1-262178ee941e)

在这个静态方法中，如果 attention_mask 为全 1（例如 batch_size 为 1，不需要做 padding 的情况），会返回 True，导致 _update_causal_mask 方法返回 None，将 LlamaModel.forward 方法中的 causal_mask 错误地设为 None：
![_ignore_causal_mask_sdpa](https://github.com/user-attachments/assets/4646a0d8-9ac2-4780-bfea-deff1e5347c8)

## 解决方法
在 from_pretrained 中添加一个参数 attn_implementation="eager"

**Comment by GCYZSL** [comment]
谢谢您指出的bug！可不可以问一下，具体是在哪一行添加attn_implementation="eager"呢？谢谢您的帮助！

**Comment by zyc140345** [comment]
比如说 mola_training_instruction.py 的 132 行
```
model = LlamaForCausalLM_d.from_pretrained(
    base_model,
    config=config,
    load_in_8bit=load_in_8bit,
    torch_dtype=torch.float16,
    device_map=device_map,
)
```
改成
```
model = LlamaForCausalLM_d.from_pretrained(
    base_model,
    config=config,
    load_in_8bit=load_in_8bit,
    torch_dtype=torch.float16,
    device_map=device_map,
    attn_implementation="eager",
)
```
其他同理

**Comment by GCYZSL** [comment]
谢谢你！最近就更新！！！超级感谢！
